src/main/java/com/minecolonies/coremod/client/render/TileEntityInfoPosterRenderer.java

```diff
@@ -0,0 +1,212 @@
+package com.minecolonies.coremod.client.render;
+
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.blockout.Log;
+import com.minecolonies.coremod.blocks.ModBlocks;
+import com.minecolonies.coremod.tileentities.TileEntityInfoPoster;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.GuiUtilRenderComponents;
+import net.minecraft.client.model.ModelSign;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.VertexBuffer;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.world.World;
+import net.minecraftforge.client.model.IModel;
+import net.minecraftforge.client.model.ModelLoader;
+import net.minecraftforge.client.model.ModelLoaderRegistry;
+import net.minecraftforge.client.model.pipeline.LightUtil;
+import net.minecraftforge.common.model.IModelState;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import org.lwjgl.opengl.GL11;
+
+import java.util.List;
+
+@SideOnly(Side.CLIENT)
+public class TileEntityInfoPosterRenderer extends TileEntitySpecialRenderer<TileEntityInfoPoster>
+{
+    private static final ResourceLocation SIGN_TEXTURE = new ResourceLocation(Constants.MOD_ID, "textures/blocks/info/info_poster.png");
+
+    /**
+     * The ModelSign instance for use in this renderer
+     */
+    private IBakedModel model = null;
+
+    @Override
+    public void renderTileEntityAt(final TileEntityInfoPoster te, final double x, final double y, final double z, final float partialTicks, final int destroyStage)
+    {
+        if(model == null)
+        {
+            model = loadModel();
+        }
+
+        final World world = te.getWorld();
+        final IBlockState state = world.getBlockState(te.getPos());
+        final BlockPos pos = te.getPos();
+        final IBlockState actualState = state.getBlock().getActualState(state, world, pos);
+
+        int k = te.getBlockMetadata();
+        float f2 = 0.0F;
+
+        if (k == 2)
+        {
+            f2 = 180.0F;
+        }
+
+        if (k == 4)
+        {
+            f2 = 90.0F;
+        }
+
+        if (k == 5)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)